app/models/reservation.rb

```diff
@@ -141,6 +141,7 @@ def approved?
   def flagged?(flag)
     # checks to see if the given flag is set
     # you must pass the symbol for the flag
+    return unless FLAGS.key? flag
```

Comment:
I think I would rather return `nil` than `false` because it differentiates between a invalid flag and a valid flag that is not set.